Angular - Frontend/src/app/app.component.ts

```diff
@@ -0,0 +1,153 @@
+import {Component, OnInit} from '@angular/core';
+import {HttpEventType, HttpHeaders, HttpResponse} from '@angular/common/http';
+import {UploadFileService} from './Services/upload-file.service';
+import {DownloadFileService} from './Services/download-file.service';
+import {DomSanitizer} from '@angular/platform-browser';
+import {saveAs} from 'file-saver';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent implements OnInit {
+  title = 'File-Upload-Save';
+  selectedFiles: FileList;
+  currentFileUpload: File;
+  currentFileDownload: File;
+  progress: { percentage: number } = {percentage: 0};
+  selectedFile = null;
+  changeImage = false;
+  formData: FormData;
+
+  constructor(private uploadService: UploadFileService,
+              private downloadService: DownloadFileService,
+              private sanitizer: DomSanitizer) {
+  }
+
+  ngOnInit() {
+  }
+
+  upload() {
+    this.resetForm();
+    // this.progress.percentage = 0;
+    this.currentFileUpload = this.selectedFiles.item(0);
+    this.uploadService.pushFileToStorage(this.currentFileUpload).subscribe(event => {
+      if (event.type === HttpEventType.UploadProgress) {
+        // this.progress.percentage = Math.round(100 * event.loaded / event.total);
+      } else if (event instanceof HttpResponse) {
+        alert('File Successfully Uploaded');
+      } else if{
+        // kiểm tra nếu file đã có
+      } else
+        //thông báo
+        // alert('File Upload Fail');
+      {
+        this.selectedFiles = undefined;
+      }
+    }
+  }
+
+);
+  sessionStorage;
+.
+
+  setItem(
+
+  'filename';
+,
+  this;
+.
+  currentFileUpload;
+.
+  name;
+);
+}
+
+
+download();
```

Comment:
this function should check more other file extensions including .mp4, .mp3, .flv, ... (binary file)